lib/rubocop/ast/processed_source.rb

```diff
@@ -240,6 +254,21 @@ def create_parser(ruby_version)
           end
         end
       end
+
+      def first_token_index(source_range)
+        sorted_tokens.bsearch_index { |token| token.begin_pos >= source_range.begin_pos }
+      end
+
+      def last_token_index(source_range)
+        sorted_tokens.bsearch_index { |token| token.end_pos >= source_range.end_pos }
```

Comment:
How about `begin_pos = source_range.begin_pos` before the `bsearch_index`, as a "free" optimization, or passing that instead of `source_range`?

Reply:
Sure, I can extract it into local variable, but it seems that won't give much a difference. For linear search this probably will have noticeable effect, but for bsearch - not sure.

I think, most of the time this method will be used with `Node`s as an argument, so 3 variants of arguments looks equal to me. `source_range` looks more clear to me, but I have no strong opinion on this.

Reply:
I feel supporting `source_range` is important as it is the most generic. There might not be a `Node` with the `source_range` someone needs.

> most of the time this method will be used with Nodes

So let's accept a `source_range`, or anything that `responds_to?(:source_range)` then.